modules/tinymce/src/plugins/link/test/ts/browser/DialogFlowTest.ts

```diff
@@ -1,151 +1,137 @@
-        value: str.startsWith('#h_')
-      })),
-      sAssertInputValue('Header One', 'Text to display'),
-      TestLinkUi.sAssertContentPresence(tinyApis, {
-        'h1[id]': 0,
-        'h2[id]': 1
-      }),
-      TestLinkUi.sClickSave,
-      TestLinkUi.sAssertContentPresence(tinyApis, {
-        'h1[id]': 1
-      }),
-
-      // Check that the h1's id value is referred to by a link containing dog
-      Chain.asStep(TinyDom.fromDom(editor.getBody()), [
-        NamedChain.asChain([
-          NamedChain.direct(NamedChain.inputName(), UiFinder.cFindIn('h1'), 'h1'),
-          NamedChain.direct('h1', Chain.mapper((h1) => Attribute.get(h1, 'id')), 'h1-id'),
-          NamedChain.bundle((obj) => UiFinder.findIn(obj[NamedChain.inputName()], `a[href="#${obj['h1-id']}"]:contains("Header One")`))
-        ])
-      ])
-    ]);
-
-    const testChangingUrlValueWithKeyboard = Log.step('TBA', 'Link: With Keyboard',
-      testChangingUrlValueWith(GeneralSteps.sequence([
-        Keyboard.sKeydown(doc, Keys.enter(), { })
-      ]))
-    );
-
-    const testChangingUrlValueWithMouse = Log.step('TBA', 'Link: With Mouse',
-      testChangingUrlValueWith(Mouse.sClickOn(TinyDom.fromDom(document.body), '.tox-collection__item')
-      )
-    );
-
-    const testChangingUrlValueManually = Log.stepsAsStep('TBA', 'Link: Change urlinput value manually', [
-      tinyApis.sSetContent('<h1>Something</h2>'),
-      tinyApis.sSetSelection([ 0, 0 ], ''.length, [ 0, 0 ], 'Something'.length),
-      TestLinkUi.sOpenLinkDialog(tinyUi),
-
-      FocusTools.sSetActiveValue(doc, 'http://www.tiny.cloud'),
-      TestLinkUi.sAssertDialogContents({
-        href: 'http://www.tiny.cloud',
-        text: 'Something',
-        title: '',
-        target: ''
-      }),
-      TestLinkUi.sClickSave,
-      TestLinkUi.sAssertContentPresence(tinyApis, {
-        a: 1
-      })
-    ]
-    );
-
-    Pipeline.async({}, [
-      TestLinkUi.sClearHistory,
-      testChangingAnchorValue,
-      testChangingUrlValueWithKeyboard,
-      testChangingUrlValueWithMouse,
-      testChangingUrlValueManually,
-      TestLinkUi.sClearHistory
-    ], onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.link.DialogFlowTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: 'link',
     toolbar: 'link',
-    theme: 'silver',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  }, [ Plugin, Theme ]);
+
+  before(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  after(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  const pAssertInputValue = async (editor: Editor, expected: string, group: string) => {
+    const input = await TestLinkUi.pFindInDialog(editor, 'label:contains("' + group + '") + input');
+    const value = UiControls.getValue(input);
+    assert.equal(value, expected, 'Checking input value');
+  };
+
+  // FIX: Dupe
+  const pAssertUrlStructure = async (editor: Editor, expected: (s, str, arr) => any) => {
```

Comment:
```suggestion
  const pAssertUrlStructure = async (editor: Editor, expected:ApproxStructure.Builder<StructAssert>) => {
```
Note: `StructAssert` isn't imported in Agar above, so better to apply this locally.